types/src/transaction/transaction.rs

```diff
@@ -146,6 +150,25 @@ pub struct AssetMintOutput {
     pub amount: Option<U256>,
 }
 
+/// Parcel transaction type.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum InnerTransaction {
+    General(Transaction),
+    AssetWrapCCC {
+        network_id: NetworkId,
+        shard_id: ShardId,
+        parcel_hash: H256,
+        output: AssetWrapCCCOutput,
+    },
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AssetWrapCCCOutput {
+    pub lock_script_hash: H160,
+    pub parameters: Vec<Bytes>,
+    pub amount: U256,
```

Comment:
@joojis What about asset composition? Does it allow infinite quantity?

Reply:
What is that mean? infinite quantity?

Inputs of the composition can be `U256` since `Asset`'s amount is U256.

Reply:
Because the current `AssetCompose` transaction reuses `AssetMintOutput`, the type of `amount` is `Option<U256>`.

Reply:
Currently, all the AssetCompose transactions verify that `amount == 1`.

https://github.com/CodeChain-io/codechain/blob/bbd0f62c70c9ec79214521ce140e03a16e55c809/types/src/transaction/transaction.rs#L271

Reply:
Does it mean that the amount of composed asset is always 1?
If that how about add new output type that has no amount?

Reply:
Yes.

And like this?
```
AssetComposeOutput {
    pub lock_script_hash: H160,
    pub parameters: Vec<Bytes>,
}
```
Sounds fair to me.

Reply:
Also sounds fair to me. Those outputs are quite different from the others. But this one is not considered in this PR.

Reply:
Okay. Let's do it in the other PR.